File: IntersectionofTwoArraysII/iotaii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1,nums2 = sorted(nums1),sorted(nums2)
        result = []
        p1,p2 = 0,0
        while p1<len(nums1) and p2<len(nums2):
            if nums1[p1] == nums2[p2]:
                result.append(nums1[p1])
                p1+=1
                p2+=1
            elif nums1[p1]>nums2[p2]:p2+=1
            else: p1+=1
        return result

File: IntersectionofTwoArraysII/iotaii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        ArrayList<Integer> result = new ArrayList<Integer>();
        for(int i:nums1){
            if(map.containsKey(i)) map.put(i,map.get(i)+1);
            else map.put(i,1);
        }
        for(int j:nums2){
            if(map.containsKey(j) && map.get(j)>=1){
                result.add(j);
                map.put(j,map.get(j)-1);
            }
        }
        int i =0;
        int[] re = new int[result.size()];
        for(int n:result){
            re[i++] = n;
        }
        return re;
    }
}